apps/snitch_core/lib/core/data/model/product.ex

```diff
@@ -315,6 +335,29 @@ defmodule Snitch.Data.Model.Product do
     end)
   end
 
+  # TODO This needs to be replaced and we need a better system to identify
+  # the type of product.
+  @spec is_parent_product(String.t()) :: true | false
+  def is_parent_product(product_id) when is_binary(product_id) do
+    query =
```

Comment:
@jyotigautam we have `product.parent_variation` which getts the variation associated with parent. If it is nill then product is parent. else is child

Reply:
I use `product.parent_variation.parent_product` in elasticsearch index

Reply:
```elixir
defp parent_or_standalone_product(%{parent_variation: nil} = product), do: product
defp parent_or_standalone_product(product), do: product.parent_variation.parent_product`
```